README.md

```diff
@@ -275,6 +276,56 @@ If you have used the `--redux` generator option, you will notice the familiar ad
 
 Note the organizational paradigm of "bundles". These are like application domains and are used for grouping your code into webpack bundles, in case you decide to create different bundles for deployment. This is also useful for separating out logical parts of your application. The concept is that each bundle will have it's own Redux store. If you have code that you want to reuse across bundles, including components and reducers, place them under `/client/app/lib`.
 
+#### Multiple React Components on a Page with One Store
+You may wish to have 2 React components share the same the Redux store. For example, if your navbar is a React component, you may want it to use the same store as your component in the main area of the page. You may even want multiple React components in the main area, which allows for greater modularity. In addition, you may want this to work with Turbolinks to minimize reloading the JavaScript.
+
+Suppose the Redux store is called `appStore`, and you have 3 React components that each need to connect to a store: `Navbar`, `Comments`, and `Blogs`.
+
+You will need to make function that can create the store you will be using for all components and register it via the `registerStoreCreator` method. Note, this is a **storeCreator**, meaning that it is a function that takes props and returns a store:
+
+```
+ReactOnRails.registerStoreCreator({
+  appStoreCreator
+});
+```
+
+When registering your component with React on Rails, you can get the store via `ReactOnRails.getStore`:
+
+```js
+// getStore will initialize the store if not already initialized, so creates or retrieves store
+const appStore = ReactOnRails.getStore("appStore");
+return (
+  <Provider store={appStore}>
+    <NonRouterCommentsContainer />
+  </Provider>
+);
+```
+
+From your Rails view, you can use the provided helper `initialize_store(store_name, props)` to create a fresh version of the store (because it may already exist if you came from visiting a previous page).
+
+**app/views/layouts/application.html.erb**
+```erb
+...
+<% initialize_store("appStore", props) %>;
```

Comment:
@justin808 lol this was a rhetorical question! You can't pass in page-specific props in a layout file without some type of yield block! Alternatively we can recommend calling the `initialize_store` method from the view, and just make sure to always pass in navbar props.